File: java/src/main/java/com/dddeurope/niuz/newsroom/NewsroomService.java
package com.dddeurope.niuz.newsroom;

import com.dddeurope.niuz.commands.CommandHandler;
import com.dddeurope.niuz.events.*;

public class NewsroomService implements CommandHandler<SubmitArticle> {
    private final AuthorRepository authors;
    private final ArticleRepository articles;
    private final EventPublisher publisher;

    public NewsroomService(AuthorRepository authors, ArticleRepository articles, Topic topic, EventPublisher publisher) {
        this.authors = authors;
        this.articles = articles;
        this.publisher = publisher;
        topic.subscribe(ContractSigned.class, this::onboard);
    }

    private void onboard(ContractSigned event) {
        authors.save(new Author(event.getAuthorId(), event.getAuthorName()));
    }

    @Override
    public boolean canHandle(Object command) {
        return command instanceof SubmitArticle;
    }

    @Override
    public void handle(SubmitArticle command) {
        articles.save(new Article(command.getArticleId(), command.getAuthorId(), command.getHeadline()));
        publisher.publish(new ArticleSubmitted(command.getAuthorId(), command.getHeadline()));
    }

    public void publish(String articleId) {
        Article article = articles.getByArticleId(articleId);
        Author author = authors.getByAuthorId(article.getAuthorId());

        publisher.publish(new ArticlePublished(author.getId(), author.getName(), article.getHeadline()));
    }
}

File: dotnet/niuz.application/newsroom/NewsRoomService.cs
using niuz.application.commands;
using niuz.application.events;

namespace niuz.application.newsroom
{
    public class NewsRoomService : ICommandHandler<SubmitArticle>
    {
        private readonly IAuthorRepository authors;
        private readonly IArticleRepository articles;
        private readonly IEventPublisher publisher;

        public NewsRoomService(IAuthorRepository authors, IArticleRepository articles, ITopic topic, IEventPublisher publisher)
        {
            this.authors = authors;
            this.articles = articles;
            this.publisher = publisher;
            topic.Subscribe<ContractSigned>(Onboard);
        }

        private void Onboard(ContractSigned @event)
        {
            authors.Save(new Author(@event.AuthorId, @event.AuthorName));
        }

        public void Publish(string articleId)
        {
            var article = articles.GetByArticleId(articleId);
            var author = authors.GetByAuthorId(article.AuthorId);

            publisher.Publish(new ArticlePublished(author.Id, author.Name, article.Headline));
        }

        public void Handle(SubmitArticle command)
        {
            articles.Save(new Article(command.ArticleId, command.AuthorId, command.Headline));
            publisher.Publish(new ArticleSubmitted(command.AuthorId, command.Headline));
        }
    }
}